source/common/http/http2/codec_impl.h

```diff
@@ -230,6 +239,7 @@ class ConnectionImpl : public virtual Connection, protected Logger::Loggable<Log
     bool pending_receive_buffer_high_watermark_called_ : 1;
     bool pending_send_buffer_high_watermark_called_ : 1;
     bool reset_due_to_messaging_error_ : 1;
+    absl::string_view details_;
```

Comment:
This assumes that whatever was used to initialize `details_` has static duration. It is right now. However I would sleep better if this was `std::string` and not think about someone calling `setDetails` with parameter of automatic storage duration.

Reply:
Yan,

I'll be happy to follow your advice, but I want to point out that I copied this pattern from the equivalent http1 code.  Does this change your advice?  (E.g., I could change the http1 code as well.)

Reply:
No it does not. I do not think this is a safe pattern.

Reply:
Thanks.  Fixed in 5e610eab4